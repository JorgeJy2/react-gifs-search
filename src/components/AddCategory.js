import React,{useState} from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');
    
    const handleInputChange = ({target: {value}})=> {
        // console.log(value);
        setinputValue(value)
    };


    const handleSubmit = (e)=> {
       e.preventDefault();
    //    console.log('Handle submit');
       
       if(inputValue.trim().length > 2){
        //    console.log(e);

           setCategories(categories =>  [inputValue,...categories]);
           setinputValue('');
       }
    };


    return (
        <form onSubmit={handleSubmit}>
            {/* <p> Add Category</p> */}
            <input 
                placeholder="Search"
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
