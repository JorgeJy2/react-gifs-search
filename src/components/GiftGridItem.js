
import React from 'react'
import PropTypes from 'prop-types';

export const GiftGridItem = ( {url,title} ) => {
    return (
        <div className="card animate__animated animate__fadeInDown">
           <img src={url} alt={title}/>
           <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

