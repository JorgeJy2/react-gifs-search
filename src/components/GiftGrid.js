import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from './hooks/UseFetchGifs';
import { GiftGridItem } from './GiftGridItem';

export const  GiftGrid = ({category}) => {

    const {data: images,loading} = useFetchGifs(category);   
    
    return (
        <>
        <p className="category">{category}</p>
        {loading && <p id="message-loading">Loading....</p>}
        <div className="card-grid">
                {
                    images?.map((img) => (
                        <GiftGridItem 
                            key={img.id}
                            {...img}
                            />
                    ))
                }
        </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
};
