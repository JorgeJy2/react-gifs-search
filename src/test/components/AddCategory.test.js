import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('test add category', ()=> {
    
    const setCategories = jest.fn();
    let wraper = shallow(<AddCategory setCategories={setCategories} />);    

    beforeEach(()=> {
        jest.clearAllMocks();
        wraper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should view', () => {

        expect(wraper).toMatchSnapshot();
    })

    test('should change input', () => {
        const input = wraper.find('input');
        const value = 'hola mundo';
        input.simulate('change', {
             target: {value   }
        }); 
    });

    test('should not clean text', ()=> {
        wraper.find('form').simulate('submit', {
            preventDefault(){}
        });
        expect(setCategories).not.toBeCalled();
    });
    
    test('should call setCategories and clean input', () =>{
        const input = wraper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', { target:  { value } });
        
        wraper.find('form').simulate('submit', { preventDefault(){} });
        
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function))    
        
        // console.log(wraper.find('input').prop('value'));

        expect( wraper.find('input').prop('value')).toBe('');

    });

});