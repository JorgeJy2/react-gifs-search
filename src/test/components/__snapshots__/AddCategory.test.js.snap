// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test add category should view 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
